File: Greedy/Python/Maximum Matrix Sum.py
# Problem link: https://leetcode.com/problems/maximum-matrix-sum/description
# Time: O(n * n)
# Space: O(n * n)
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        count = total = 0
        track = float('inf')

        for row in matrix:
            for num in row:
                if num < 0:
                    count += 1
                track = min(track, abs(num))
                total += abs(num)
        
        return total if count % 2 == 0 else total - track * 2

File: Greedy/Java/Maximum Matrix Sum.java
// Problem link: https://leetcode.com/problems/maximum-matrix-sum/description
// Time: O(n * n)
// Space: O(n * n)
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        int track = Integer.MAX_VALUE, count = 0;
        long result = 0;

        for(int[] arr : matrix) {
            for(int num : arr) {
                if(num < 0) {
                    count++;
                }
                result += Math.abs(num);
                track = Math.min(track, Math.abs(num));
            }
        }

        return count % 2 == 0 ? result : result - track * 2;
    }
}
